### PR TITLE
feat(topology): add show labels in pools

### PR DESCRIPTION
**Pull request by sinhaashish**
Get pool 
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get pool pool-on-node-1 -n openebs --show-labels
 ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED  LABELS
 pool-on-node-1  aio:///dev/sdb?uuid=b324aa79-c8b3-4859-a6e8-1a51fbc5d944  true     node-1-237668  Online  10GiB     0 B        10GiB      0 B        topology-key=topology-value
```
List pools 
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get pools  -n openebs --show-labels
 ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED  LABELS
 pool-on-node-2  aio:///dev/sdb?uuid=8045fe61-ae51-429b-9c73-4f8f60c42d44  true     node-2-237668  Online  10GiB     0 B        10GiB      0 B
 pool-on-node-0  aio:///dev/sdb?uuid=4b12de79-fa6c-4bfd-9ed0-4f5f57cfeb4c  true     node-0-237668  Online  10GiB     0 B        10GiB      0 B        zone=us-east-1
 pool-on-node-1  aio:///dev/sdb?uuid=b324aa79-c8b3-4859-a6e8-1a51fbc5d944  true     node-1-237668  Online  10GiB     0 B        10GiB      0 B        topology-key=topology-value
 ```
List Pools with node filter
```
[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get pools  -n openebs --show-labels --node  node-2-237668
 ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED  LABELS
 pool-on-node-2  aio:///dev/sdb?uuid=8045fe61-ae51-429b-9c73-4f8f60c42d44  true     node-2-237668  Online  10GiB     0 B        10GiB      0 B

[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get pools  -n openebs --show-labels --node  node-1-237668
 ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED  LABELS
 pool-on-node-1  aio:///dev/sdb?uuid=b324aa79-c8b3-4859-a6e8-1a51fbc5d944  true     node-1-237668  Online  10GiB     0 B        10GiB      0 B        topology-key=topology-value

[nix-shell:~/code/rust/mayastor-extensions]$ ./target/debug/kubectl-mayastor get pools  -n openebs --show-labels --node  node-0-237668
 ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED  LABELS
 pool-on-node-0  aio:///dev/sdb?uuid=4b12de79-fa6c-4bfd-9ed0-4f5f57cfeb4c  true     node-0-237668  Online  10GiB     0 B        10GiB      0 B        zone=us-east-1
```

### CODE REVIEW

File: control-plane/plugin/src/resources/pool.rs [hunk 1]
```diff
@@ -279,3 +328,80 @@ impl Label for Pool {
...
+    pub(crate) fn get_pool_label_list(pool: &openapi::models::Pool) -> Vec<String> {
+        let mut pools_labels: Vec<String> = vec![];
+
+        match &pool.spec {
+            Some(spec) => match &spec.labels {
+                Some(ds) => {
+                    pools_labels = ds
+                        .iter()
+                        // Dont return the created_by_dsp label for the gets
+                        .filter(|(key, _)| *key != &"openebs.io/created-by")
```
Comment (Abhinandan-Purkait) [inline control-plane/plugin/src/resources/pool.rs]: Use the created by key from constants

**Review by Abhinandan-Purkait — CHANGES_REQUESTED** [review]

File: control-plane/plugin/src/resources/pool.rs (hunk 1 above)
Comment (sinhaashish) [reply control-plane/plugin/src/resources/pool.rs]: done

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/plugin/src/resources/pool.rs [hunk 2]
```diff
@@ -133,7 +166,17 @@ impl ListWithArgs for Pools {
             },
             None => true,
         });
-        utils::print_table(output, pools);
+
+        let pools_display = PoolDisplayLabels::new_pools(pools.clone(), args.show_labels());
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]: Doesn't seem to be used afterwards
```suggestion
        let pools_display = PoolDisplayLabels::new_pools(pools, args.show_labels());
```

File: control-plane/plugin/src/resources/pool.rs [hunk 3]
```diff
@@ -279,3 +330,81 @@ impl Label for Pool {
         Ok(())
     }
 }
+
+/// The PoolDisplayLabels structure is responsible for controlling the display formatting of Pool
+/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
+/// object is serialised, only the `inner` object is represented.
+#[derive(Serialize, Debug)]
+pub struct PoolDisplayLabels {
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]: This is not just displaying the labels.
Also in the future we might add other info.

```suggestion
/// The PoolDisplay structure is responsible for controlling the display formatting of Pool
/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
/// object is serialised, only the `inner` object is represented.
#[derive(Serialize, Debug)]
pub struct PoolDisplay {
```

File: control-plane/plugin/src/resources/pool.rs [hunk 4]
```diff
@@ -279,3 +330,81 @@ impl Label for Pool {
...
+    /// Create a new `PoolDisplay` instance from a vector of pools.
+    pub(crate) fn new_pools(pools: Vec<openapi::models::Pool>, show_labels: bool) -> Self {
+        Self {
+            inner: pools,
+            show_labels,
+        }
+    }
+
+    /// Get a list of pool labels.
+    pub(crate) fn get_pool_label_list(pool: &openapi::models::Pool) -> Vec<String> {
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]:
```suggestion
    pub(crate) fn pool_label_list(pool: &openapi::models::Pool) -> Vec<String> {
```

File: control-plane/plugin/src/resources/pool.rs [hunk 5]
```diff
@@ -279,3 +330,81 @@ impl Label for Pool {
...
+    fn create_rows(&self) -> Vec<Row> {
+        let mut rows = vec![];
+        for pool in self.inner.iter() {
+            let mut row = pool.create_rows();
+            if self.show_labels {
+                let labelstring = PoolDisplayLabels::get_pool_label_list(pool).join(", ");
+                // Add the pool labels to each row.
+                row[0].add_cell(Cell::new(&labelstring));
+            }
+            rows.push(row[0].clone());
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]:
```suggestion
            let mut row = pool.row();
            if self.show_labels {
                let labelstring = PoolDisplayLabels::get_pool_label_list(pool).join(", ");
                // Add the pool labels to each row.
                row.add_cell(Cell::new(&labelstring));
            }
            rows.push(row);
```

File: control-plane/plugin/src/resources/pool.rs [hunk 6]
```diff
@@ -68,6 +71,27 @@ impl GetHeaderRow for openapi::models::Pool {
     }
 }
 
+#[derive(Debug, Clone, clap::Args)]
+/// Arguments used when getting a pool.
+pub struct GetPoolArgs {
+    /// Id of the pool
+    pool_id: PoolId,
+    /// Show the labels of the pool
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]:
```suggestion
/// Arguments used when getting a pool.
#[derive(Debug, Clone, clap::Args)]
pub struct GetPoolArgs {
    /// Id of the pool.
    pool_id: PoolId,
    /// Show the labels of the pool.
```

File: control-plane/plugin/src/resources/pool.rs [hunk 7]
```diff
@@ -84,6 +108,10 @@ pub struct GetPoolsArgs {
     /// Pools must satisfy all of the specified label constraints.
     #[clap(short = 'l', long)]
     selector: Option<String>,
+
+    /// Show the labels of the pool
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/pool.rs]:
```suggestion
    /// Show the labels of the pool.
```

**Review by tiagolobocastro — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors cancel

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**